src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1306,3 +1370,25 @@ int sd_dhcp_server_set_relay_target(sd_dhcp_server *server, const struct in_addr
         server->relay_target = *address;
         return 1;
 }
+
+int sd_dhcp_server_set_relay_agent_circuit_id(
+                sd_dhcp_server *server,
+                const char *agent_circuit_id) {
+        assert_return(server, -EINVAL);
+
+        if (relay_agent_suboptions_length(agent_circuit_id, server->agent_remote_id) > 255)
```

Comment:
Hm, please consider the situation such that
```
sd_dhcp_server_set_relay_agent_circuit_id(server, "(200 of 'a')");
sd_dhcp_server_set_relay_agent_remote_id(server, "(100 of 'b')");
sd_dhcp_server_set_relay_agent_circuit_id(server, "(100 of 'c')");
```
Then, `server->agent_circuit_id` should be `"ccc..."`, and `server->agent_remote_id` should be `"bbb..."`. However, because of this check, the above `sd_dhcp_server_set_relay_agent_remote_id()` fails.

How about to merge the two setters like the following:
```
int sd_dhcp_server_set_relay_agent_information(sd_dhcp_server *server, const char *circuit_id, const char *remote_id) {
    ...
    if (relay_agent_information_length(circuit_id, remote_id) > 255)
        return -EINVAL;

    ...
}
```